src/odemis/driver/hamamatsurx.py

```diff
@@ -0,0 +1,2144 @@
+                    if last_error_code:
+                        # log the last error code received before timeout
+                        logging.error("Last error code for function %s before timeout was %s with message %s."
+                                  % (last_error_fct, last_error_code, last_error_msg))
+                    raise util.TimeoutError("No answer received after %s sec for command %s."
+                                            % (timeout, command.encode('string_escape')))
+
+                try:
+                    error_code, rfunc, rargs = int(response[0]), response[1], response[2:]
+                except Exception as msg:
+                    raise IOError("Received response, which is not according to the known protocol."
+                                  "Error message was %s" % msg)
+
+                # check if the response corresponds to the command sent before
+                # the response corresponding to a command always also includes the command name
+                if rfunc.lower() == func.lower():  # fct name not case sensitive
+                    logging.debug("Hamamatsu streak camera RemoteEx response: %s." % response)
+                else:
+                    # save the last error message and code in case we don't receive any other response before timeout
+                    last_error_code = error_code
+                    last_error_fct = rfunc
+                    last_error_msg = rargs
+                    logging.debug("Hamamatsu streak camera RemoteEx response not as expected. "
+                                  "Will wait some more time.")
+                    continue  # continue listening to receive the correct response for the sent command or timeout
+
+                if error_code:  # != 0, response corresponds to command, but an error occurred
+                    logging.error(RemoteExError(error_code))
+                    raise RemoteExError(error_code)
+                else:  # successfully executed command and return message
+                    return rargs
+
+    def readCommandResponse(self):
+        """
+        This method runs in a separate thread and continuously listens for messages returned from
+        the device via the commandport IP socket.
+        The messages are made available either on .queue_command_responses (for the standard responses)
+        or .queue_img (for messages related to the images).
+        """
+        try:
+            responses = ""  # received data not yet processed
+
+            while self.should_listen:
+                try:
+                    returnValue = self._commandport.recv(4096)  # buffersize should be small value of power 2 (4096)
+                except socket.timeout:
+                    # when socket timed out (receiving no response)
+                    logging.debug("Timeout on the socket, will wait for more data packages.")
+                    continue
+
+                responses += returnValue
```

Comment:
I'd recommand to log the received value from RemoteEx right here, as is. This can be handy in case something goes wrong in the interpretation of a response.